modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/components/OrderableTable.tsx

```diff
@@ -0,0 +1,322 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton, {ClayButtonWithIcon} from '@clayui/button';
+import ClayEmptyState from '@clayui/empty-state';
+import ClayLayout from '@clayui/layout';
+import ClayTable from '@clayui/table';
+import classNames from 'classnames';
+import {ManagementToolbar} from 'frontend-js-components-web';
+import fuzzy from 'fuzzy';
+import React, {useRef, useState} from 'react';
+import {DndProvider, useDrag, useDrop} from 'react-dnd';
+import {HTML5Backend} from 'react-dnd-html5-backend';
+
+import Search from './Search';
+
+import '../../css/OrderableTable.scss';
+
+const FUZZY_OPTIONS = {
+	post: '</strong>',
+	pre: '<strong>',
+};
+
+interface OrderableTableRowInterface {
+	fields: Array<{
+		label: string;
+		name: string;
+	}>;
+	index: number;
+	item: any;
+	onOrderChange: Function;
+	query: string;
+}
+
+const OrderableTableRow = ({
+	fields,
+	index,
+	item,
+	onOrderChange,
+	query,
+}: OrderableTableRowInterface) => {
+	const ref = useRef<HTMLTableRowElement>(null);
```

Comment:
I _think_ that we want descriptive ref names but not sure 🤷

Reply:
Yeah, you're right. This block of code, related to application of `react-dnd`, I mostly left unchanged from implementation in Objects.